src/lapack/backends/rocsolver/rocsolver_scope_handle.hpp

```diff
@@ -57,7 +57,11 @@ class RocsolverScopedContextHandler {
     // will be fixed when SYCL-2020 has been implemented for Pi backend.
     template <typename T, typename U>
     inline T get_mem(U acc) {
+#ifdef SYCL_IMPLEMENTATION_ONEAPI
```

Comment:
Here's where it is defined: https://github.com/intel/llvm/blob/c3ca0941c9fa6b6371a1defd32d338d57d6b7b54/sycl/include/sycl/define_vendors.hpp#L10.

That's a good point, the reason I introduced the `ifdef` was that I was assuming that `backend::hip` was working on another sycl implementation, but as you mention there is also a reference to `sycl::backend::ext_oneapi_hip`.

I'll encapsulate all hip backend reference with the same `ifdef`  :+1:

Reply:
Right, so DPC++ is `sycl::backend::ext_oneapi_hip` while HipSYCL is `sycl::backend::hip`.
Adding the `ifdef` to all references will make sure that all sycl implementations can compile.

Reply:
Ahhh, thanks!
Currently the rocSOLVER backend only supports the LLVM* compiler (according to the table on the readme page). However, it seems like a good idea to encapsulate for the future, if/once the hipSYCL compiler is supported. I believe the history for why the backend::hip was used is because that worked pre-SYCL2020. The cuSOLVER/rocSOLVER backends haven't yet been updated to work with the SYCL2020 changes (see issue #360 ).

Reply:
Oh right, I was under the impression that it also supported the hipSYCL compiler, thus the `ifdef`.
Should we just remove the `ifdef` and replace `hip` with `ext_oneapi_hip` then?